pylint/config/arguments_manager.py

```diff
@@ -0,0 +1,107 @@
+
+"""Arguments manager class used to handle command-line arguments and options."""
+
+import argparse
+import configparser
+from typing import TYPE_CHECKING, Dict, List
+
+from pylint.config.argument import _Argument
+from pylint.config.exceptions import UnrecognizedArgumentAction
+from pylint.config.utils import _convert_option_to_argument
+
+if TYPE_CHECKING:
+    from pylint import checkers
+
+
+class _ArgumentsManager:
+    """Arguments manager class used to handle command-line arguments and options."""
+
+    def __init__(self) -> None:
+        self.namespace = argparse.Namespace()
+        """Namespace for all options."""
+
+        self._arg_parser = argparse.ArgumentParser(prog="pylint")
+        """The command line argument parser."""
+
+        self._argument_groups_dict: Dict[str, argparse._ArgumentGroup] = {}
+        """Dictionary of all the argument groups."""
+
+    def _register_options_provider(self, provider: "checkers.BaseChecker") -> None:
+        """Register an options provider and loads its defaults."""
+        # pylint: disable-next=fixme
+        # TODO: Do something own_group parameter (see OptionsManagerMixIn.register_options_provider)
+        for opt, optdict in provider.options:
+            argument = _convert_option_to_argument(opt, optdict)
+            self._add_arguments_to_parser(provider.name, argument)
+
+        # pylint: disable-next=fixme
+        # TODO: Do something with option provider priorities (see OptionsManagerMixIn.register_options_provider)
+        # pylint: disable-next=fixme
+        # TODO: Do something with option groups within optdicts (see OptionsManagerMixIn.register_options_provider)
+
+        # pylint: disable-next=fixme
+        # TODO: Investigate performance impact of loading default arguments on every call
+        self._load_default_argument_values()
+
+    def _add_arguments_to_parser(self, section: str, argument: _Argument) -> None:
+        """Iterates over all argument sections and add them to the parser object."""
+        try:
+            section_group = self._argument_groups_dict[section]
+        except KeyError:
+            section_group = self._arg_parser.add_argument_group(title=section)
+            self._argument_groups_dict[section] = section_group
+        self._add_parser_option(section_group, argument)
+
+    @staticmethod
+    def _add_parser_option(
+        section_group: argparse._ArgumentGroup, argument: _Argument
+    ) -> None:
+        """Add an argument."""
+        if argument.action == "store":
+            section_group.add_argument(
+                *argument.name,
+                action=argument.action,
+                default=argument.default,
+                type=argument.type,  # type: ignore[arg-type] # incorrect typing in typeshed
+                help=argument.help,
+                metavar=argument.metavar,
+                choices=argument.choices,
+            )
+        else:
+            raise UnrecognizedArgumentAction
```

Comment:
May I ask if there is a reason we can't just permit other actions like `store_true` and send them along? As in, is this something a plugin writer could take advantage of? (I'm used to `store_true` and the docs/tutorial suggest it also...)

Reply:
We should at some point. For now we only need `store` as logging only has `store` options (that's why I chose to implement there first).

The `_Argument` class also does not support of parameters that can be passed to `add_argument()`. I want to add these on a if-needed basis when I convert more checkers.